server/src/server.ts

```diff
@@ -194,9 +194,16 @@ export default class BashServer {
             ? `${symbolKindToDescription(symbol.kind)} defined in ${path.relative(
                 currentUri,
                 symbol.location.uri,
+              )}\n\n${this.analyzer.commentsAbove(
```

Comment:
We should add `\n\n` if no comment is found (i.e. it returns null).

Reply:
Do you mean that `commentsAbove` should return `\n\n` instead of null if nothing is found?
Or do you mean that we should not add the `\n\n` between the `defined in X` part and the `commentsAbove` part?

Reply:
We should not add `\n\n` if there is no comment. I missed a negation in my first comment.

Reply:
sounds good. What do you think of something like this:

```
`${symbolKindToDescription(symbol.kind)} defined in ${path.relative(
    currentUri,
    symbol.location.uri,
)}${this.analyzer.commentsAbove(
    symbol.location.uri,
    symbol.location.range.start.line,
)}`
```

and if `commentsAbove` does not find any comments above, then it can return an empty string. So that way the tooltip simply displays: `X defined in Y` (and similarly for the section below where `X defined on line Y`)

Reply:
If it always returns a string then you would need to return the new lines from the function. As we want two new lines in case a comment is found. 

Instead the formatting should be done at display time, formatting is not a concern of the commentsAbove function. So I recommend just returning a plain string or a null. 

In case of a null then don’t display, else add two new lines before the string.

Reply:
Ok. I just wanted to minimize the logic from within the template string because it already is quite messy. How about this:

```
// a helper function that either returns the comment string with two new lines
// or an empty string if not found
const getCommentsAbove = (symbol) => {
    const commentString = this.analyzer.commentsAbove(
        symbol.location.uri,
        symbol.location.range.start.line,
    )
    if (commentString) {
        return `\n\n${commentString}`
    }

    // return empty string if commentString is null
    return ''
}


`${symbolKindToDescription(symbol.kind)} defined in ${path.relative(
    currentUri,
    symbol.location.uri,
)}${getCommentsAbove(symbol)}`
```

Reply:
It totally makes sense to extract to a variable and not do it in the template string. 

A string would be fine instead of a function, and the code comment shouldn’t be needed - the code reads nicely. 

```JavaScript

const comment = this.analyzer.commentsAbove(
        symbol.location.uri,
        symbol.location.range.start.line,
 )
const commentString = comment ? `\n\n${comment}` : ''
```

Reply:
:+1: 

sounds good

Reply:
actually looking at it more closely, I think the best way is to use a helper function because the template string is located in a "one-line" filter and map expression, and I want to minimize the amount of code changed in that function